graphics/partio/Portfile

```diff
@@ -30,3 +32,29 @@ depends_build-append \
                     port:doxygen \
                     port:swig-python
 depends_lib         port:zlib
+
+pre-test {
+    if {![variant_isset tests]} {
+        ui_error "'tests' variant must be activated to enable test support"
+        error "Please enable the 'tests' variant and try again"
+    }
+}
```

Comment:
this is interesting. I wonder if this would work:
```
pre-test {
  default_variants-append tests
}
```

Reply:
or whatever the right syntax is to add a default variant :> .. . somehow I think it's a bit different than the usual.

Reply:
That's a really good idea @kencu. I would've never thought of that.

Reply:
Let me try it out and report back.

Reply:
It doesn't work. I suspect it's because by the time the build process gets to the `pre-test` phase, it's already too late to be specifying a variant. However, I wonder if it's possible for the portfile to discover right at the beginning of a build what port command the user issued, i.e. that they issued a `port test` (vs. a `port install`) command?